conans/client/conan_api.py

```diff
@@ -428,11 +428,15 @@ def config_get(self, item):
     def config_set(self, item, value):
         config_parser = ConanClientConfigParser(self._client_cache.conan_conf_path)
         config_parser.set_item(item, value)
+        self._client_cache._conan_config = None
+        # self._client_cache = get_client_cache(get_conan_user_home(), self._user_io.out)
```

Comment:
![why these commented lines?](https://4loc.files.wordpress.com/2009/11/lines_of_code.jpg)

Reply:
:D :D 

Also 2 things:
- Access to protected => better avoid
- There are more things to invalidate in the ClientCache

I suggest an ``invalidate()`` or ``reset()`` method in ClientCache that takes care of cleaning cached values. Have a look to fields in __ init __